pydantic_numpy/model.py

```diff
@@ -152,12 +163,20 @@ def dump(
 
         return dump_directory_path
 
-    def _dump_numpy_split_dict(self) -> tuple[dict, dict]:
-        ndarray_field_to_array = {}
-        other_field_to_value = {}
-
-        for k, v in self.model_dump(exclude_unset=True).items():
-            if isinstance(v, np.ndarray):
+    @staticmethod
+    def _dump_numpy_split_dict(model: "NumpyModel") -> tuple[dict[str, np.ndarray], dict[str, Any]]:
+        ndarray_field_to_array = dict[str, np.ndarray]()
```

Comment:
Don't parametrize the dict, but add type hints to the variable itself, that way we stay close to PEP8:
```suggestion
        ndarray_field_to_array: dict[str, np.ndarray] = {}
```
Apply this to the others please